File: apps/mobile/modules/maher-watch-bridge/ios/MaherWatchBridgeModule.swift
import ExpoModulesCore

public class MaherWatchBridgeModule: Module {
  public func definition() -> ModuleDefinition {
    Name("MaherWatchBridge")

    OnCreate {
      MaherWatchSession.shared.activate()
    }

    AsyncFunction("publishContext") { (payload: [String: Any]) in
      MaherWatchSession.shared.publishContext(payload)
    }
    .runOnQueue(.main)

    AsyncFunction("publishToken") { (accessToken: String, sessionEpoch: Int) in
      MaherWatchSession.shared.publishToken(accessToken, sessionEpoch: sessionEpoch)
    }
    .runOnQueue(.main)

    AsyncFunction("clear") {
      MaherWatchSession.shared.clear()
    }
    .runOnQueue(.main)
  }
}

File: apps/mobile/modules/maher-watch-bridge/src/MaherWatchBridgeModule.ts
import { NativeModule, requireOptionalNativeModule } from 'expo-modules-core';

export type MaherWatchBridgeNativeModule = NativeModule & {
  publishContext(payload: Record<string, unknown>): Promise<void>;
  publishToken(accessToken: string, sessionEpoch: number): Promise<void>;
  clear(): Promise<void>;
};

export default requireOptionalNativeModule<MaherWatchBridgeNativeModule>('MaherWatchBridge');
